### PR TITLE
Dirc lut improvement

### PR DESCRIPTION
**Pull request by jrstevenjlab**

### CODE REVIEW

**Comment by gluex** [comment]
Build status for this pull request: FAILURE
 
 Build log: [/work/halld/pull_request_test/halld_recon^dirc_lut_improvement/make_dirc_lut_improvement.log](https://halldweb.jlab.org/pull_request_test/halld_recon^dirc_lut_improvement/make_dirc_lut_improvement.log)
 Build report: [/work/halld/pull_request_test/halld_recon^dirc_lut_improvement/report_dirc_lut_improvement.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^dirc_lut_improvement/report_dirc_lut_improvement.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^dirc_lut_improvement

**Comment by gluex** [comment]
Build status for this pull request: FAILURE
 
 Build log: [/work/halld/pull_request_test/halld_recon^dirc_lut_improvement/make_dirc_lut_improvement.log](https://halldweb.jlab.org/pull_request_test/halld_recon^dirc_lut_improvement/make_dirc_lut_improvement.log)
 Build report: [/work/halld/pull_request_test/halld_recon^dirc_lut_improvement/report_dirc_lut_improvement.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^dirc_lut_improvement/report_dirc_lut_improvement.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^dirc_lut_improvement

**Comment by gluex** [comment]
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^dirc_lut_improvement/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^dirc_lut_improvement/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^dirc_lut_improvement/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^dirc_lut_improvement/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^dirc_lut_improvement/make_dirc_lut_improvement.log](https://halldweb.jlab.org/pull_request_test/halld_recon^dirc_lut_improvement/make_dirc_lut_improvement.log)
 Build report: [/work/halld/pull_request_test/halld_recon^dirc_lut_improvement/report_dirc_lut_improvement.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^dirc_lut_improvement/report_dirc_lut_improvement.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^dirc_lut_improvement

File: src/libraries/DIRC/DDIRCLut.cc [hunk 1]
```diff
@@ -68,16 +72,27 @@ bool DDIRCLut::brun(JEventLoop *loop) {
...
-	for(int i=0; i<12; i++) {
-	  dRotationX[i] = -0.2134*TMath::DegToRad(); // +pitch // -0.00219;
-	  dRotationY[i] =  0.0963*TMath::DegToRad(); // -yaw   //  0.00181
-	  dRotationZ[i] = -0.0355*TMath::DegToRad(); // -roll  // -0.00062;
+	// get cherenkov angle corrections from CCDB
+	vector <double> thetac_offsets_bar(5184);
+	if(loop->GetCalib("/DIRC/thetac_offsets_bar", thetac_offsets_bar)) 
+	  jout << "Can't find requested /DIRC/thetac_offsets_bar in CCDB for this run!" << endl;
+	
+	for(int ibar=0; ibar<48; ibar++) {
```
Comment (sdobbs) [inline src/libraries/DIRC/DDIRCLut.cc:80]: Would be nice to define these numbers in DDIRCGeometry and get their values from there.  See for example the constant factors in DFCALGeometry.

File: src/libraries/DIRC/DDIRCLut.cc [hunk 2]
```diff
@@ -249,10 +264,9 @@ vector<pair<double,double>> DDIRCLut::CalcPhoton(const DDIRCPmtHit *locDIRCHit,
 	if((box == 0 && channel < dMaxChannels) || (box == 1 && channel >= dMaxChannels)) 
 		return locDIRCPhotons;
 
-	int pmt = channel/64;
 	double rotX = 0, rotY = 0, rotZ=0;
 	int xbin = (int)(posInBar.X()/5.0) + 19;
-	if(bar>=0 && bar<=23 && xbin>=0 && xbin<=39) {
+	if(bar>=0 && bar<=47 && xbin>=0 && xbin<=39) {
```
Comment (sdobbs) [inline src/libraries/DIRC/DDIRCLut.cc:269]: Same for geometry factors like these...

File: src/plugins/Analysis/dirc_hists/DEventProcessor_dirc_hists.cc [hunk 3]
```diff
@@ -321,12 +334,18 @@ jerror_t DEventProcessor_dirc_hists::evnt(JEventLoop *loop, uint64_t eventnumber
 					  }
 					  
 					  // fill histograms for candidate photons in timing cut
-					  if(fabs(locDeltaT) < 5.0) {
+					  if(fabs(locDeltaT) < 3.0) {
```
Comment (sdobbs) [inline src/plugins/Analysis/dirc_hists/DEventProcessor_dirc_hists.cc:337]: Do you want to be able to override these cuts via command line parameters?

**Comment by jrstevenjlab** [comment]
Thanks for the comments Sean.  I tried to clean up the use of bar/pmt/etc constants from the geometry class.  I'm submitting a followup PR now.

**Pull request merged**